src/main/kotlin/org/rust/cargo/project/settings/ui/RustProjectSettingsPanel.kt

```diff
@@ -40,43 +48,49 @@ class RustProjectSettingsPanel(
         val explicitPathToStdlib: String?
     )
 
-    override fun dispose() {}
+    override fun dispose() {
+        Disposer.dispose(pathToToolchainComboBox)
+    }
 
     private val versionUpdateDebouncer = UiDebouncer(this)
 
-    private val pathToToolchainField = pathToDirectoryTextField(this,
-        "Select directory with cargo binary") { update() }
+    private val pathToToolchainComboBox = RsToolchainPathChoosingComboBox().apply {
```

Comment:
nit: Let's better pass `DocumentListener` into `RsToolchainPathChoosingComboBox` constructor and set up it inside the object intself instead of exposing `pathTextField`

Reply:
Done.